packages/discovery-react-components/src/components/SearchFacets/components/ShowMore/ShowMoreButton.tsx

```diff
@@ -0,0 +1,49 @@
+import React, { FC } from 'react';
+import { Button } from 'carbon-components-react';
+import { Messages } from 'components/SearchFacets/messages';
+
+interface ShowMoreButtonProps {
+  /**
+   * onClick handler to call for the Show more button
+   */
+  onClick: () => void;
+  /**
+   * Suffix to use for the button's data-testid
+   */
+  idSuffix: string;
+  /**
+   * If the button is expanded to show more or collapsed
+   */
+  isCollapsed: boolean;
+  /**
+   * If the button text will be for opening a modal or expanding the list
+   */
+  isShowAllMessage: boolean;
+  /**
+   * i18n messages for the component
+   */
+  messages: Messages;
+}
+
+export const ShowMoreButton: FC<ShowMoreButtonProps> = ({
+  onClick,
+  idSuffix,
+  isCollapsed,
+  isShowAllMessage,
+  messages
+}) => {
+  const showMessage = () => {
```

Comment:
Why does this need to be a function?

Reply:
@anibapat requested I break it out of the render into a function for readability.

Reply:
It's not a huge deal, but you should just be able to set the value of the message within the component function itself:

```tsx
let message = messages.collapsedFacetShowLessText;

if (isShowAllMessage) {
  message = messages.collapsedFacetShowAllText;
} else if (isCollapsed) {
  message = messages.collapsedFacetShowMoreText;
}

return (
  <Button kind="ghost" size="small" onClick={onClick} data-testid={`show-more-less-${idSuffix}`}>
    {message}
  </Button>
);
```

Reply:
Oh right wasn't thinking clearly, can simply set a message as Nick said.